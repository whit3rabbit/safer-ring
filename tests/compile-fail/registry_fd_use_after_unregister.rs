use safer_ring::Registry;

fn main() {
    let mut registry = Registry::new();
    let registered_fd = registry.register_fd(0).unwrap();
    
    // Unregister the fd
    registry.unregister_fd(registered_fd).unwrap();
    
    // Try to use the fd after unregistration - should not compile
    println!("FD index: {}", registered_fd.index()); //~ ERROR: borrow of moved value
}