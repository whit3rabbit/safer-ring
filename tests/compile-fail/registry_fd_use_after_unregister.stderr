error[E0382]: borrow of moved value: `registered_fd`
  --> tests/compile-fail/registry_fd_use_after_unregister.rs:11:30
   |
5  |     let registered_fd = registry.register_fd(0).unwrap();
   |         ------------- move occurs because `registered_fd` has type `RegisteredFd`, which does not implement the `Copy` trait
...
8  |     registry.unregister_fd(registered_fd).unwrap();
   |                            ------------- value moved here
...
11 |     println!("FD index: {}", registered_fd.index()); //~ ERROR: borrow of moved value
   |                              ^^^^^^^^^^^^^ value borrowed here after move
